### PR TITLE
BaSP-TG-43: Modal Form shared component

### PR DESCRIPTION
**Pull request by Javier-Cervera**
- Modal implemented on Employees Add and Edit functionalities

### CODE REVIEW

File: src/Components/Shared/ModalForm/modal-form.module.css [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  background-color: rgba(0, 0, 0, 0.5);
+}
+
+.modal {
+  display: flex;
+  flex-direction: column;
+  align-self: center;
+  margin: 0 auto;
+  width: 700px;
+  height: 550px;
```
[the remark this answers is not in the document]
Comment (Javier-Cervera) [reply src/Components/Shared/ModalForm/modal-form.module.css:21]: Thanks for the feedback!

File: src/Components/Employees/SuccessModal/index.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ const SuccessModal = (props) => {
 
   const closeBothModals = () => {
     props.closeModal();
-    props.closeFormModal();
+    props.closeModalForm();
   };
 
   return props.successResponse.error === true ? (
```
[the remark this answers is not in the document]
Comment (Javier-Cervera) [reply src/Components/Employees/SuccessModal/index.js:13]: Nice

**Comment by tinchorova** [comment]
## Testing Results
Tested locally on https://alexis-trackgenix-app-git-feature-tg-43-share-427909-basp-m2022.vercel.app/

**Test Results: Failed** ⚠️

Bugs found:
1. Both add and edit modals should not have a scroll bar
image
image
2. When on edit modal, user shouldn't be allowed to scroll the page
3. While on edit or add modal, clicking outside of the modal, holding the click and moving the cursor up or down should not select the elements of the form.
image
image

**Review by tinchorova — CHANGES_REQUESTED** [review]
Pleas refer to: https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/13#issuecomment-1149261932

**Comment by tinchorova** [comment]
## Testing Results
Tested locally on https://alexis-trackgenix-app-git-feature-tg-43-share-427909-basp-m2022.vercel.app/employees

**Test Results: Failed** ⚠️

Bugs found:
1. When selecting a text inside an input holding click button and releasing the click outside the modal it closes the window **(Clicking inside the modal and releasing the click outside of it, closes the modal)** dot in image shows the release position and circle represents clicking position
image
2. When clicking outside the modal and going inside it closes the window. A solution for this could be to close the window on click instead of on release. Dot in image shows the release position and circle represents clicking position
image

**Review by tinchorova — CHANGES_REQUESTED** [review]
Please refer to: https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/13#issuecomment-1151298613

**Comment by tinchorova** [comment]
## Test Results 

Tested locally on https://alexis-trackgenix-app-git-feature-tg-43-share-427909-basp-m2022.vercel.app/employees

**Test Results: Failed** ⚠️

Bugs found:

1. There should be more space between layers and inputs, and between each group of layer-input. A solution for this could be to create a second cloumn so that there is more space to order the elements.
image
2. The exit (X) button and font are too big. You could use figma's model as an example for this. Reference image below.
image
image

**Review by tinchorova — CHANGES_REQUESTED** [review]
Please refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/13#issuecomment-1151447652

**Comment by LeonelListro** [comment]
Test Results
Tested locally on https://alexis-trackgenix-app-git-feature-tg-43-share-427909-basp-m2022.vercel.app/employees

Test Results: Failed ⚠️

Bugs found:

1. The cross to go back on the success modal of the add and edit modal is too big.

cross

2. The content of the password field should be hidden.

image

**Review by LeonelListro — APPROVED** [review]
Please refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/13#issuecomment-1151611940

**Pull request merged**